### ISSUE TITLE
JRuby doesn't start on ppc64le when loading environment

### ISSUE THREAD

**Issue by mew2057** [jruby/jruby#5725]
### Environment
* JRuby bundled in  logstash 7.0.0  ( jar is `jruby-complete-9.2.7.0.jar`)
* Linux h1 4.14.0-29.el7a.ppc64le #1 SMP Mon Jan 22 16:26:32 EST 2018 ppc64le ppc64le ppc64le GNU/Linux

Other relevant info you may wish to add:
* Application : Logstash

### Expected Behavior

* When running the `environment.rb` JRuby should load all modules without exiting execution.

### Actual Behavior

* When the JRuby Application (logstash) is loaded a LoadError is thrown by ffi/ffi. The bubbled error message is somewhat vague (to a user ignorant to JRuby).

```
[2019-05-03T10:41:38,701][ERROR][org.logstash.Logstash    ] 
java.lang.IllegalStateException: Logstash stopped processing because of an error: 
(LoadError) load error: ffi/ffi -- java.lang.NullPointerException: null
```

### Resolution

I was able to trace the bug to `jruby/lib/ruby/stdlib/ffi/platform/powerpc64-linux/`. It looks 
as though the `platform.conf` file was not created for this platform. Copying the `types.conf` file to `platform.conf` appears to resolve the problem.

**Comment by ryanbreed** [comment]
gonna +1 on this - have the same issue in `ruby/lib/ruby/stdlib/ffi/platform/arm-linux`
Same result - failure to launch logstash on raspbian stretch `4.19.33-rockchip #5.77 armv7l`

**Comment by headius** [comment]
@mew2057 @ryanbreed Can you turn that into a PR? Better yet, if we can regenerate these files for these platforms then we'll know they're accurate.

**Comment by jetnet** [comment]
the workaround mentioned above does not help unfortunately on:
```
$ uname -a
Linux jetson 4.9.140-tegra #1 SMP PREEMPT Wed Mar 13 00:32:22 PDT 2019 aarch64 aarch64 aarch64 GNU/Linux
```

**Comment by Trevor159** [comment]
This solution also did not work for me 
`Linux odroidxu4 4.14.111-odroidxu4 #2 SMP PREEMPT Wed May 8 17:30:01 CEST 2019 armv7l armv7l armv7l GNU/Linux`

**Comment by headius** [comment]
To those folks getting a similar error on a platform OTHER than PPC: I think it would be best if you opened a separate issue. I got confused about your comments because they do not have anything to do with PowerPC.

**Comment by headius** [comment]
@jetnet I have confirmed on my aarch64 env that JRuby loads ok, and also loads "ffi/ffi" correctly. If you are still having problems, please open a new bug specific to your environment.

@Trevor159 Closing the loop here...we chatted on Matrix, and you said you'd try to get a standalone reproduction (without Logstash). I do not have an armv7l environment to test with, so I'm relying on your results for now. Do open a new issue if you can reproduce with JRuby directly. A command line like `jruby -rffi -e 1` should fail if it's a JRuby problem.

**Comment by glassman81** [comment]
> gonna +1 on this - have the same issue in `ruby/lib/ruby/stdlib/ffi/platform/arm-linux`
> Same result - failure to launch logstash on raspbian stretch `4.19.33-rockchip #5.77 armv7l`
> same fix resolved the issue - copying types.conf to platform.conf

@ryanbreed I'm also using armv7l. WHat exactly do you mean by copying types.conf to platform.conf?

Did you copy types.conf, creating a duplicate of it as platform.conf into the 'arm-linux folder? I'm soory if this seems elementary, I'm just a little confused.

**Comment by kares** [comment]
@glassman81 that was done at https://github.com/jruby/jruby/commit/5e16c801dcc24d0fbd334d7d7b2d29cc9413e225 (so its in JRuby 9.2.8.0)
... if you're having issues please make sure you're using latest JRuby and open a separate issue (for armv7l) so we can track it. 

this one was left-open for someone to confirm JRuby >= 9.2.8.0 is working on *powerpc64-linux* arch.

**Comment by glassman81** [comment]
> @glassman81 that was done at [5e16c80](https://github.com/jruby/jruby/commit/5e16c801dcc24d0fbd334d7d7b2d29cc9413e225) (so its in JRuby 9.2.8.0)
> ... if you're having issues please make sure you're using latest JRuby and open a separate issue (for armv7l) so we can track it.
> 
> this one was left-open for someone to confirm JRuby >= 9.2.8.0 is working on _powerpc64-linux_ arch.

That's fine. I didn't read the thread carefully enough to understand I was meant to re-jar the file. I can confirm it works on arm and arm64 this way, so I should think it also works in powerpc64. Hopefully that info helps anyone considering using JRuby on those platforms.

**Comment by besawn** [comment]
@kares @headius I am a colleague of @mew2057, and I am attempting to validate the fix for this issue on ppc64le. My knowledge of both jruby and logstash is limited, can you confirm if this is a valid test for this issue:

I downloaded the following jar files directly from https://www.jruby.org/files/downloads:
`jruby-complete-9.2.7.0.jar` (the original version where @mew2057 first discovered the problem), `jruby-complete-9.2.8.0.jar` (the version that should contain the fix)
`jruby-complete-9.2.9.0.jar` (current latest) 

I then ran the following on my test machine:
```
$ uname -a
Linux compute01 4.14.0-49.el7a.ppc64le #1 SMP Wed Mar 14 13:58:40 UTC 2018 ppc64le ppc64le ppc64le GNU/Linux

$ cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.5 (Maipo)

$ ls
jruby-complete-9.2.7.0.jar  jruby-complete-9.2.8.0.jar  jruby-complete-9.2.9.0.jar

$ java -jar jruby-complete-9.2.7.0.jar -rffi -e 1
LoadError: load error: ffi/ffi -- java.lang.NullPointerException: null
  require at org/jruby/RubyKernel.java:987
  require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
   <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/ffi.rb:1
  require at org/jruby/RubyKernel.java:987
   (root) at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1

$ java -jar jruby-complete-9.2.8.0.jar -rffi -e 1
LoadError: load error: ffi/ffi -- java.lang.NullPointerException: null
  require at org/jruby/RubyKernel.java:987
  require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
   <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/ffi.rb:1
  require at org/jruby/RubyKernel.java:987
   (root) at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:1

$ java -jar jruby-complete-9.2.9.0.jar -rffi -e 1
LoadError: load error: ffi/ffi -- java.lang.NullPointerException: null
  require at org/jruby/RubyKernel.java:978
  require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
   <main> at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/ffi.rb:1
  require at org/jruby/RubyKernel.java:978
  require at uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
```
On the same test machine, I also ran the same command using the patched version of `jruby-complete-9.2.7.0` that exists within our logstash installation. This was patched using the method @mew2057 described above.
```
$ pwd
/usr/share/logstash/logstash-core/lib/jars
   
$ ls jruby-complete-9.2.7.0.jar 
jruby-complete-9.2.7.0.jar

$ java -jar jruby-complete-9.2.7.0.jar -rffi -e 1
```

It appears to me that all three versions I tested from jruby.org exhibit the problem, but the patched version does not. Can you confirm whether the tests above are valid from your perspective, and if not, give some advice on how I can improve them?

Thanks for your assistance!
